accepted/2020/workloads/workload-manifest.md

```diff
@@ -0,0 +1,430 @@
+
+Note that `extends` is functionally a dependency system and a way to factor out common sets of packages from workloads. By analogy to package managers such as apt-get and NuGet, workloads are metapackages that only permit unversioned dependencies and packs are packages that are only installable transitively.
+
+## Pack Definitions
+
+Pack definitions take the following form:
+
+| Key | Type | Value | Required |
+|--|--|--|--|
+| `kind` | string | Type of the pack. Valid values are `sdk`, `framework`, `library`, `template`, `tool`. | Yes |
+| `version` | string | Version of the pack. | Yes |
+| `alias-to` | object | Alias this definition to the provided package ID | No |
+
+The `alias-to` key allows aliasing this pack definition to a different underlying pack ID on each host platform. The keys are host platform RIDs, and the values are pack IDs. The `"*"` platform ID may be used to alias a pack on all platforms.
+
+For example, an SDK pack might contain a compiler that's a native executable and hence depends on the host platform:
+
+```json
+"foo.sdk.compiler": {
+    "version": "1.0",
+    "kind": "sdk",
+    "alias-to": {
+        "osx-x64": "foo.sdk.compiler.mac",
+        "win-x64": "foo.sdk.compiler.windows"
+    }
+}
+```
+
+Workload definitions and SDK imports would refer to this as `"foo.sdk.compiler"`, but on Windows the pack that gets installed and resolved would be `"foo.sdk.compiler.windows"`. The `"foo.sdk.compiler.windows"` pack would be present on disk and its ID would be visible in resolved paths, but otherwise transparent and not able to be used directly.
+
+Another use of aliases is to install multiple version bands of the same pack by synthesizing new versioned pack IDs:
+
+```json
+"foo.framework": {
+    "version": "2.0.4",
+    "kind": "framework",
+    "alias-to": {
+        "*": "foo.framework",
+    }
+},
+"foo.framework.1": {
+    "version": "1.3.2",
+    "kind": "framework",
+    "alias-to": {
+        "*": "foo.framework",
+    }
+},
+```
+
+A workload or framework reference that referenced `"foo.framework"` would get `"foo.framework"` version `2.0.4`, while a workload or framework reference that referenced `"foo.framework.1"` would get `"foo.framework"` version `1.3.2`,
+
+## Example Platform SDK Manifest
+
+Here is a *hypothetical* example manifest. It's not prescriptive but demonstrates concepts an patterns that can be used by platform implementations.
+
+```json5
+{
+    "version": 5,
+    "workloads": {
+        // this is a dev workload that would typically be installed by a developer getting
+        // started with this platform. it's composed of several smaller build workloads
```

Comment:
This surprised me. Would this workload not be used by someone more experienced?

Reply:
They might restore a project they already had or install a more complete workload knowing the optional features they'd need? Added a clarification.